### ISSUE TITLE
Issue with DataParallelModel

### ISSUE THREAD

**Issue by youansheng** [zhanghang1989/PyTorch-Encoding#133]
https://github.com/zhanghang1989/PyTorch-Encoding/blob/c2cb2aab69d5d276fbcb847fb8277c1a52947661/experiments/segmentation/train.py#L130

Why the outputs are not gathered in DataParallelModel? Thanks!

**Comment by yinjunbo** [comment]
@zhanghang1989, Hi, does encoding support multi-task losses?  When using single GPU, the return of my model is a dictionary with each key/value indicating a specific task loss.  I tried the `encoding.parallel.DataParallelModel`, and the return  is a list of dictionary. The length of the list is equal to the number of devices.  Then I apply `torch.nn.parallel._functions.Gather` to the output list to integrate the losses from different devices in a unified GPU device. But the experimental results are not good as the  `torch.nn.DataParallel` with the same batchsize. So Is there something I missed?
